common/core/overlaylayer.h

```diff
@@ -185,9 +185,9 @@ struct OverlayLayer {
   }
 
   void SetProtected(bool isProtected) {
-    if (isProtected && type_ == kLayerVideo)
+    if (isProtected && (type_ == kLayerVideo || type_ == kLayerProtected))
```

Comment:
is the 'protected' only for video layer? If so kLayerProtectedVideo may be better than kLayerProtected.

Reply:
yes, it is only for video now.

Reply:
@YuanjunHuang 
Let's keep this PR just for logic fix not anything else.
If you propose better name or refactory, you can give another PR

Reply:
ok sure.